plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeRegisterTableProcedureTest.java

```diff
@@ -0,0 +1,273 @@
+        assertUpdate(format("DROP TABLE %s", tableName));
+    }
+
+    @Test
+    public void testRegisterTableWithComments()
+    {
+        String tableName = "test_register_table_with_comments_" + randomTableSuffix();
+
+        assertQuerySucceeds(format("CREATE TABLE %s (a, b, c) COMMENT 'my-table-comment' AS VALUES (1, 'INDIA', true), (2, 'USA', false)", tableName));
+        assertThat(query(format("SELECT * FROM %s", tableName)))
+                .matches("VALUES ROW(INT '1', VARCHAR 'INDIA', BOOLEAN 'true'), ROW(INT '2', VARCHAR 'USA', BOOLEAN 'false')");
+
+        String tableLocation = getTableLocation(tableName);
+        // Drop table from hive metastore and use the same table name to register again with the metadata
+        dropTableFromMetastore(tableName);
+
+        assertQuerySucceeds("CALL system.register_table (CURRENT_SCHEMA, '" + tableName + "', '" + tableLocation + "')");
+        assertThat(getTableComment(tableName)).isEqualTo("my-table-comment");
+
+        assertUpdate(format("DROP TABLE %s", tableName));
+    }
+
+    @Test
+    public void testRegisterTableWithShowCreateTable()
+    {
+        String tableName = "test_register_table_with_show_create_table_" + randomTableSuffix();
+
+        assertQuerySucceeds(format("CREATE TABLE %s AS SELECT 1 as a, 'INDIA' as b, true as c", tableName));
+        assertThat(query(format("SELECT * FROM %s", tableName)))
+                .matches("VALUES ROW(INT '1', VARCHAR 'INDIA', BOOLEAN 'true')");
+
+        String tableLocation = getTableLocation(tableName);
+        String showCreateTableOld = (String) computeActual("SHOW CREATE TABLE " + tableName).getOnlyValue();
+
+        // Drop table from hive metastore and use the same table name to register again with the transaction logs
+        dropTableFromMetastore(tableName);
+
+        assertQuerySucceeds(format("CALL %s.system.register_table('%s', '%s', '%s')", CATALOG_NAME, SCHEMA, tableName, tableLocation));
+        String showCreateTableNew = (String) computeActual("SHOW CREATE TABLE " + tableName).getOnlyValue();
+
+        assertThat(showCreateTableOld).isEqualTo(showCreateTableNew);
+        assertUpdate(format("DROP TABLE %s", tableName));
+    }
+
+    @Test
+    public void testRegisterTableWithDifferentTableName()
+    {
+        String tableName = "test_register_table_with_different_table_name_old_" + randomTableSuffix();
+
+        assertQuerySucceeds(format("CREATE TABLE %s AS SELECT 1 as a, 'INDIA' as b, true as c", tableName));
+        assertThat(query(format("SELECT * FROM %s", tableName)))
+                .matches("VALUES ROW(INT '1', VARCHAR 'INDIA', BOOLEAN 'true')");
+
+        String showCreateTableOld = (String) computeActual("SHOW CREATE TABLE " + tableName).getOnlyValue();
+        String tableLocation = getTableLocation(tableName);
+        // Drop table from hive metastore and use the same table name to register again with the transaction logs
```

Comment:
This comment does not apply - probably a copy-paste consequence.

Reply:
Reword the comment.